005_hackerrank_challenges/001_introduction/008_reverse_a_list/solution.hs

```diff
@@ -1,4 +1,49 @@
-rev l = go l
-    where go [] = []
-          go [x] = [x]
-          go (x:xs) = (go xs) ++ [x]
+#! /usr/bin/env nix-shell
+#! nix-shell ../../../004_an_updated_nix_shell_file/shell.nix  -i runghc
+
+-- run it locally by doing `cat sample-input.txt | ./solution.hs`
+
+import Data.List (foldl', foldr)
+import Control.Monad
+
+rev :: [a] -> [a]
+rev [] = []
+rev [x] = [x]
+ -- the problem here is you are appending
```

Comment:
This was transformed to get rid of the inner `go` helper since it was exactly the same as the outer function.